src/tree.rs

```diff
@@ -1147,4 +1270,34 @@ mod tree_tests {
         let sub_root = tree.find_subtree_root_between_ids(&root_id, &node_4_id); //invert for None
         assert_eq!(sub_root, None);
     }
+
+    #[test]
+    fn test_swap_sub_trees() {
+        let mut tree = Tree::new();
```

Comment:
Any thoughts here?  @Drakulix no worries if you can't think of anything else to test. just trying to be thorough.